### PR TITLE
Fixes #33565 - speed-up kickstart on air gapped infra where repos are unreachable

### PR DESCRIPTION
**Pull request by pandrieux**
speed-up kickstart on air gapped infra where repos are unreachable

on air gapped infra, installing subscription-manager won't be possible as-is since internet repo is unreachable.

I'm simply checking RPM is installed avoiding to wait for timeout.

### CODE REVIEW

File: app/views/unattended/provisioning_templates/snippet/redhat_register.erb [hunk 1]
```diff
@@ -110,10 +110,14 @@ snippet: true
   echo "Starting the subscription-manager registration process"
 
   <% if !atomic %>
-    if [ -f /usr/bin/dnf ]; then
-      dnf -y install subscription-manager
-    else
-      yum -t -y install subscription-manager
+    # Avoid timeout accessing unreachable repo on air gapped infrastructure,
+    #  assuming subscription-manager is installed in custom packages section.
+    if [ -z "$(rpm -qa subscription-manager)" ]; then
```
Comment (ekohl) [inline app/views/unattended/provisioning_templates/snippet/redhat_register.erb]: Should it instead use `rpm --query --quiet subscription-manager` and rely on the exit code?
```suggestion
    if ! rpm --query --quiet subscription-manager ; then
```

**Comment by ekohl** [comment]
Could you squash both commits and use the right commit message format? Including `Fixes #33565` allows our automation to work.

**Comment by pandrieux** [comment]
thanks you @ekohl , I think that is ok now.

**Comment by ekohl** [comment]
ok to test

**Review by ekohl — DISMISSED** [review]
I started CI, if that's green it can be merged.

**Comment by ekohl** [comment]
My apologies, I forgot to check back. Now there's a merge conflict. Could you rebase this patch to resolve the conflict?

**Comment by ShimShtein** [comment]
@pandrieux Can you please rebase the PR?

Also: I see there are other yum install commands in this file, or in other snippets that could be called. I suppose you will need to wrap them with a similar check too.

**Comment by pandrieux** [comment]
@ShimShtein 
Sorry it took me some time as I was ooo for a while.
I just rebased the code.
I can't tell for the over yum install commands. In my env, I wrapped only this one.

**Review by ShimShtein — CHANGES_REQUESTED** [review]
Other yum commands are invoked in some specific cases. That's why you didn't see them manifesting in your setup.
For example the one [here](https://github.com/theforeman/foreman/blob/dfe4a66d6d0a0122c8ce69bc5f53abc7646e3339/app/views/unattended/provisioning_templates/snippet/redhat_register.erb#L127) will be invoked if you set `syspurpose*` parameters.
and others in this snippet would be manifested if content management (Katello) is enabled for this host.

I think we need to count in all these cases.

**Comment by pandrieux** [comment]
I understand your point. 
I read the .erb, I'm adding wrapper for `subscription-manager-syspurpose` because it seems to be the same case.
I don't see any other invocation that require the wrapper. Once `subscription-manager*` is installed and registration is done, repositories will be available.

**Review by ShimShtein — APPROVED** [review]

**Comment by ShimShtein** [comment]
Merged, thanks @pandrieux !

**Pull request merged**